src/commands/functions/create.js

```diff
@@ -196,23 +195,76 @@ const pickTemplate = async function () {
 
 const DEFAULT_PRIORITY = 999
 
-/* get functions dir (and make it if necessary) */
-const ensureFunctionDirExists = function (context, flags, config) {
-  const functionsDir = config.build && config.build.functions
-  if (!functionsDir) {
-    context.log(`${NETLIFYDEVLOG} No functions folder specified in netlify.toml`)
-    process.exit(1)
+/**
+ * Get functions dir (and make it if necessary)
+ * @this FunctionsCreateCommand
+ * @returns {string | never} - functions directory or throws an error
+ */
+const ensureFunctionDirExists = async function () {
+  const { api, config, site } = this.netlify
+  const { log } = this
+  let functionsDirHolder = config.build && config.build.functions
+  let siteData = null
+  const siteId = site.id
+
+  // try to find functions_dir in site settings if not in user's netlify config
+  if (!functionsDirHolder) {
+    try {
+      siteData = await api.getSite({ siteId })
+    } catch (error) {
+      if (error.status === 404) {
+        error('Site not found')
+      } else {
+        error(error.message)
+      }
+    }
+
+    // https://open-api.netlify.com/#operation/getSite
+    functionsDirHolder = siteData?.build_settings?.functions_dir
   }
-  if (!fs.existsSync(functionsDir)) {
-    context.log(
+
+  if (!functionsDirHolder) {
+    log(`${NETLIFYDEVLOG} functions folder not specified in netlify.toml or app settings`)
+    const { functionsDir } = await inquirer.prompt([
+      {
+        type: 'input',
+        name: 'functionsDir',
+        message: 'Enter the path where your functions should live:',
+        default: 'netlify/functions',
+      },
+    ])
+
+    try {
+      log(`${NETLIFYDEVLOG} updating app build settings with ${chalk.magenta.inverse(functionsDirHolder)}`)
+      await api.updateSite({
+        siteId,
```

Comment:
`siteId` is not guaranteed to exist, for example when a user hasn't linked the directory to a site yet.
Should we check that use case and exit with a relevant error?

Reply:
I think we should update it if the user is explicitly entering it, it might be confusing for them to see the functions directory as "not set" if they set it as the default.